packages/api/src/index.ts

```diff
@@ -58,23 +71,46 @@ class API extends LCDClient {
     msgs: IMsg<any>[],
     account: IAccount,
     opts: { fee?: IFee, chain_id?: string, account_number?: number, sequence?: number, memo?: string } = {}) {
+    const account_number = opts.account_number
+      ? opts.account_number.toString()
+      : account.account_number.toString()
+    const chain_id = opts.chain_id || await this.fetchChainId()
+    const sequence = opts.sequence
+      ? opts.sequence.toString()
+      : account.sequence.toString()
+    const fee = opts.fee || await (async (): Promise<IFee> => {
+      const gas = await this.estimateGas(msgs, account.address, chain_id, account_number, sequence)
+      return {
+        amount: [{ denom: 'atto', amount: gas }], // amount = gas because minGasPrice=1.0
+        gas
+      }
+    })()
     return new Transaction({
-      account_number: opts.account_number
-        ? opts.account_number.toString()
-        : account.account_number.toString(),
-      chain_id: opts.chain_id || (await this.getRequest(`/node_info`)).node_info.network,
-      fee: opts.fee || {
-        amount: [{ denom: 'atto', amount: '300000' }],
-        gas: '300000'
-      },
+      account_number,
+      chain_id,
+      fee,
       memo: opts.memo || '',
       msgs: msgs,
-      sequence: opts.sequence
-        ? opts.sequence.toString()
-        : account.sequence.toString()
+      sequence,
     })
   }
 
+  async fetchChainId(): Promise<string> {
+    return (await this.getRequest(`/node_info`)).node_info.network
+  }
+
+  async estimateGas(msgs: IMsg<any>[], from: string, chain_id: string, account_number: string, sequence: string): Promise<string> {
+    const base_req = {
+      from,
+      chain_id,
+      account_number,
+      sequence: sequence,
+      gas_adjustment: '1.5',
+    } as BaseReq
+    const res = await this.postRequest('/txs/estimategas', { base_req, msgs })
```

Comment:
Based on this comment

```suggestion
    const res = await this.postRequest('/txs/simulate', { base_req, msgs })
```